src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -49,6 +54,29 @@ export async function getDataSourcesWithFields(
   return response?.savedObjects;
 }
 
+export async function handleSetDefaultDatasourceAfterDeletion(
+  savedObjects: SavedObjectsClientContract,
+  uiSettings: IUiSettingsClient
+) {
+  uiSettings.remove('defaultDataSource');
+  const listOfDataSources: DataSourceTableItem[] = await getDataSources(savedObjects);
+  if (Array.isArray(listOfDataSources) && listOfDataSources.length >= 1) {
+    const datasourceId = listOfDataSources[0].id;
+    await uiSettings.set('defaultDataSource', datasourceId);
+  }
+}
+
+export async function handleSetDefaultDatasourceDuringCreation(
+  savedObjects: SavedObjectsClientContract,
+  uiSettings: IUiSettingsClient
+) {
+  const listOfDataSources: DataSourceTableItem[] = await getDataSources(savedObjects);
+  if (Array.isArray(listOfDataSources) && listOfDataSources.length === 1) {
+    const datasourceId = listOfDataSources[0].id;
+    await uiSettings.set('defaultDataSource', datasourceId);
+  }
+}
+
```

Comment:
Thanks for this change, 

Maybe following scenario is out of scope: 

Suppose we already have dataSource created before before we support default dataSource, this line will failed to set the first dataSource as default: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6237/files#diff-fde383d0fdaaed3bb243ed36abd631c9c311596b2057f737253fc5ab11c4d5edR74

Thus I'm thinking if we can handle that scenario and also keep one function to handle set default dataSource by updating it to:

```suggestion
export async function handleSetDefaultDatasource(
  savedObjects: SavedObjectsClientContract,
  uiSettings: IUiSettingsClient
) {
  const currentDefaultDataSourceId = uiSettings.get('defaultDataSource');
  if (currentDefaultDataSourceId) {
     const targetDataSource = await getDataSourceById(currentDefaultDataSourceId, savedObjectsClient);
     if (targetDataSource.id === currentDefaultDataSourceId) {
        return;
     }
  }
  setFirstDataSourceAsDefault(savedObjectsClient, uiSettings)
}

async function setFirstDataSourceAsDefault(  
  savedObjectsClient: SavedObjectsClientContract,
  uiSettings: IUiSettingsClient
) {
  uiSettings.remove('defaultDataSource');
  const listOfDataSources: DataSourceTableItem[] = await getDataSources(savedObjects);
  if (Array.isArray(listOfDataSources) && listOfDataSources.length >= 1) {
    const datasourceId = listOfDataSources[0].id;
    await uiSettings.set('defaultDataSource', datasourceId);
  }
}
```

Just suggestions, what do you think?

Reply:
I think this is a better approach. I will have a new CR on default with picker. Do you mind if i make the relative change in the new CR.